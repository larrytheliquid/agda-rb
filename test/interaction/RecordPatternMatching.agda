module RecordPatternMatching where

record _×_ (A B : Set) : Set where
  constructor _,_
  field
    proj₁ : A
    proj₂ : B

data Unit : Set where
  unit : Unit

foo : Unit × Unit → Unit
foo (x , y) = {!!}

record Box (A : Set) : Set where
  constructor [_]
  field
    proj : A

bar : Box Unit → Unit
bar [ x ] = {!!}
