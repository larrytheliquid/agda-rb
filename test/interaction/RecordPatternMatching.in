top_command (cmd_load currentFile [])

-- Case splitting should work.
goal_command 1 cmd_make_case "x"
goal_command 0 cmd_make_case "x"
