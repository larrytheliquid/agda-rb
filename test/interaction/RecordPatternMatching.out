agda2_mode_code (agda2-status-action "")
agda2_mode_code (agda2-status-action "")
agda2_mode_code (agda2-info-action
                 "*All Goals*"
                 "?0 : Unit
?1 : Unit
")
agda2_mode_code (agda2-goals-action '(0 1))
agda2_mode_code (last . (agda2-make-case-action
                         '("bar [ unit ] = ?")))
agda2_mode_code (agda2-goals-action '(0 1))
agda2_mode_code (last . (agda2-make-case-action
                         '("foo (unit , y) = ?")))
agda2_mode_code (agda2-goals-action '(0 1))
